pkg/controllers/operation_controller.go

```diff
@@ -50,39 +50,40 @@ type OperationReconciler struct {
 	// mutex lock for engines
 	mutex *sync.Mutex
 
-	// Real implement
-	implement dataoperation.OperationReconcilerInterface
+	// Real implementBuilder
+	implementBuilder dataoperation.OperationReconcilerInterfaceBuilder
```

Comment:
I suggest using OperationInterfaceBuilder instead of OperationReconcilerInterfaceBuilder.